skimage/morphology/rolling_ball.py

```diff
@@ -0,0 +1,189 @@
+    frequently used rolling ball algorithm [1]_.
+
+    Parameters
+    ----------
+    image : ndarray
+        The last two dimensions are treated as the columns and rows of the
+        image. The operation is broadcasted along the remaining ones.
+    kernel_size: ndarray, optional
+        The length of the non-intensity vertices of the ellipsoid.
+        ``kernel_size`` must have the same dimensions as ``image``.
+    intensity_vertex : scalar, numeric, optional
+        The length of the intensity vertex of the ellipsoid.
+    has_nan: bool, optional
+        If ``False`` (default) assumes that none of the values in ``image``
+        are ``np.nan``, and uses a faster implementation.
+    num_threads: int, optional
+        The maximum number of threads to use. If ``None`` use the OpenMP
+        default value; typically equal to the maximum number of virtual cores.
+        Note: This is an upper limit to the number of threads. The exact number
+        is determined by the system's OpenMP library.
+
+    Returns
+    -------
+    filtered_image : ndarray
+        The estimated background of the image.
+
+    Notes
+    -----
+
+    - For the pixel that has its background intensity estimated (without loss
+      of generality at ``(0,0)``) the rolling ellipsoid method places an
+      ellipsoid under it and raises the ellipsoid until its surface touches the
+      image umbra at ``pos=(y,x)``. The background intensity is then estimated
+      using the image intensity at that position (``image[y, x]``) plus the
+      difference of ``intensity_vertex`` and the surface of the ellipsoid at
+      ``pos``. The surface intensity of the ellipsoid is computed using the
+      canonical ellipsis equation::
+
+            semi_spatial = kernel_size / 2
+            semi_vertex = intensity_vertex / 2
+            np.sum((pos/semi_spatial)**2) + (intensity/semi_vertex)**2 = 1
+
+    - This algorithm assumes that dark pixels correspond to the background. If
+      you have a bright background, invert the image before passing it to the
+      function, e.g., using `utils.invert`.
+    - This algorithm is sensitive to noise (in particular salt-and-pepper
+      noise). If this is a problem in your image, you can apply mild
+      gaussian smoothing before passing the image to this function.
+
+    References
+    ----------
+    .. [1] Sternberg, Stanley R. "Biomedical image processing." Computer 1
+           (1983): 22-34. :DOI:`10.1109/MC.1983.1654163`
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skimage import data
+    >>> from skimage.morphology import rolling_ellipsoid
+    >>> image = data.coins()
+    >>> background = rolling_ellipsoid(data.coins())
+    >>> filtered_image = image - background
+    """
+
+    kernel_size = np.asarray(kernel_size)
+    if not np.issubdtype(kernel_size.dtype, np.number):
+        raise ValueError(
+            "kernel_size must be convertible to a numeric array.")
```

Comment:
no, the check isn't necessary.

When I wrote the first version of this function for 2D uint8 images in pure python about a month ago (time flies!), I implemented it with proper type-checking. Back then @jni (I think) commented that skimage has the philosophy of guiding the user with good documentation instead of raising errors for inputs if possible. I received suggestions for improvements on the type-checking code (thank you again @mkcor for meticulously improving my bad spelling), so I wasn't sure if I should keep it or remove it.

Long story short, at some point I asked an open question, but never received confirmation either way. I'm happy to remove it (c'est python after all) or keep it. The same goes for the other type checks.

Reply:
I vote for removing ;)